includes/Checker/Checks/Plugin_Updater_Check.php

```diff
@@ -0,0 +1,226 @@
+		$plugin_header    = get_plugin_data( $plugin_main_file );
+		if ( ! empty( $plugin_header['UpdateURI'] ) ) {
+			$this->add_result_error_for_file(
+				$result,
+				true,
+				__( 'Plugin Updater detected. Use of the Update URI header is not helpful in plugins hosted on WordPress.org.', 'plugin-check' ),
+				'plugin_updater_detected',
+				$plugin_main_file
+			);
+		}
+	}
+
+	/**
+	 * Looks for plugin updater file and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result    The check result to amend, including the plugin context to check.
+	 * @param array        $php_files List of absolute PHP file paths.
+	 */
+	protected function look_for_updater_file( Check_Result $result, array $php_files ) {
+
+		$plugin_update_file = self::filter_files_by_regex( $php_files, '/plugin-update-checker\.php$/' );
+
+		if ( $plugin_update_file ) {
+			foreach ( $plugin_update_file as $file ) {
+				$this->add_result_error_for_file(
+					$result,
+					true,
+					sprintf(
+						/* translators: %s: The match updater file name. */
+						__( 'Plugin Updater detected. These are not permitted in WordPress.org hosted plugins. Detected: %s', 'plugin-check' ),
+						basename( $file )
+					),
+					'plugin_updater_detected',
+					$file
+				);
+			}
+		}
+	}
+
+	/**
+	 * Looks for plugin updater code in plugin files and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result    The check result to amend, including the plugin context to check.
+	 * @param array        $php_files List of absolute PHP file paths.
+	 */
+	protected function look_for_plugin_updaters( Check_Result $result, array $php_files ) {
+
+		$look_for_regex = array(
+			'#\'plugin-update-checker\'#i',
+			'#WP_GitHub_Updater#i',
+			'#WPGitHubUpdater#i',
+			'#class [A-Z_]+_Plugin_Updater#i',
+			'#updater\.\w+\.\w{2,5}#i',
+			'#site_transient_update_plugins#i',
+		);
+
+		foreach ( $look_for_regex as $regex ) {
+			$updater_file = self::file_preg_match( $regex, $php_files, $matches );
+			if ( $updater_file ) {
+				$this->add_result_error_for_file(
+					$result,
+					true,
+					sprintf(
+						/* translators: %s: The match updater string. */
+						__( 'Plugin Updater detected. These are not permitted in WordPress.org hosted plugins. Detected: %s', 'plugin-check' ),
+						esc_attr( $matches[0] )
+					),
+					'plugin_updater_detected',
+					$updater_file
+				);
+			}
+		}
+	}
+
+	/**
+	 * Looks for plugin updater routines in plugin files and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result    The check result to amend, including the plugin context to check.
+	 * @param array        $php_files List of absolute PHP file paths.
+	 */
+	protected function look_for_updater_routines( Check_Result $result, array $php_files ) {
+
+		$look_for_regex = array(
+			'#auto_update_plugin#i',
```

Comment:
Same as above, this value is case-sensitive in https://github.com/WordPress/plugin-check/blob/trunk/checks/plugin-updaters.php#L57